### PR TITLE
Add tests for property-optional

### PR DESCRIPTION
**Pull request by pshao25**
Add tests for property-optional

### CODE REVIEW

File: test/CadlRanchProjects/property-optional/Generated/BytesClient.cs [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+
+#nullable disable
+
+using System;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace property_optional
```
Comment (lirenhe) [inline test/CadlRanchProjects/property-optional/Generated/BytesClient.cs:14]: same question with this namespace

File: test/CadlRanchProjects.Tests/property-optional.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using System.Xml;
+using AutoRest.TestServer.Tests.Infrastructure;
+using Azure;
+using Models.Property.Optional;
+using NUnit.Framework;
+using property_optional;
```
Comment (lirenhe) [inline test/CadlRanchProjects.Tests/property-optional.cs:11]: Is it possible to make the namespace and class name different?

File: test/CadlRanchProjects.Tests/property-optional.cs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+namespace CadlRanchProjects.Tests
+{
+    public class property_optional : CadlRanchTestBase
```
[the remark this answers is not in the document]
Comment (pshao25) [reply test/CadlRanchProjects.Tests/property-optional.cs:15]: This is the class name of test class. I don't think we need to follow naming pattern.

File: test/CadlRanchProjects.Tests/property-optional.cs (hunk 2 above)
Comment (pshao25) [reply test/CadlRanchProjects.Tests/property-optional.cs:11]: Yes, we could change the namespace of generated SDK to Property.Optional

File: test/CadlRanchProjects.Tests/property-optional.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (lirenhe) [reply test/CadlRanchProjects.Tests/property-optional.cs:15]: Seems this is manual written so it is better to make naming consistent, shall we use PropertyOptional instead

**Comment by lirenhe** [comment]
@pshao25, could you resolve the build failure?

**Review by lirenhe — APPROVED** [review]

**Pull request merged**